docker-compose.yml

```diff
@@ -1087,6 +1088,32 @@ services:
         /arrow/ci/scripts/cpp_build.sh /arrow /build &&
         /arrow/ci/scripts/python_sdist_test.sh /arrow"
 
+  ############################ Python free-threading ##########################
+
+  ubuntu-python-313-freethreading:
```

Comment:
Should this be parameterized on the Python version just like `conda-python`?

Reply:
Eventually yes, but was thinking that we could also handle that when there are actually multiple python versions that could be used here (but also fine to already do that, of course)

Reply:
Also, while unlikely, free-threading support is experimental in 3.13 and might not even make it to 3.14. So hardcoding 3.13 makes sense to me.